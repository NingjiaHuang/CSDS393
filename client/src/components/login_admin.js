import React, { Fragment, useState } from "react";
import {Link, Redirect} from "react-router-dom";
import {toast} from "react-toastify";
import 'react-toastify/dist/ReactToastify.css'

const AdminLogin = ({setAuth}) => {
    const [inputs, setInputs] = useState({
        reg_email: "",
        password: ""
    }); 
    const {reg_email, password} = inputs;

    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value});
    };

    const onSubmitForm = async (e) => {
        e.preventDefault()
        try {
            const body = {reg_email, password}
            const response = await fetch("http://localhost:4020/auth/login/admin", {
                method: "POST",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            })
            const parseRes = await response.json();
            if(parseRes.token){
                localStorage.setItem("token", parseRes.token);
                setAuth(true)
                toast.success("Login successfully.");
            } else {
                setAuth(false);
                toast.error(parseRes);
            }

        } catch(err) {
            console.error(err.message);
        }
    }
    return (
        <Fragment>
            <h1 className="text-center my-5">Administrator Login</h1>
            <form onSubmit={onSubmitForm}>
                <label>Email: </label><input type="email" name="reg_email" placeholder="email" className="form-control my-2" value={reg_email} onChange={e => onChange(e)}/>
                <label>Password: </label><input type="password" name="password" placeholder="password" className="form-control my-2" value={password} onChange={e => onChange(e)}/>
                <button className="btn btn-success btn-block">Login</button>
            </form>
        </Fragment>
    )
}

export default AdminLogin;