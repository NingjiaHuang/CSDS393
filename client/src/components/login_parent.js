import React, { Fragment, useState } from "react";
import {Link, Redirect} from "react-router-dom";
import {toast} from "react-toastify";
import 'react-toastify/dist/ReactToastify.css'

const ParentLogin = ({setAuth}) => {
    const [inputs, setInputs] = useState({
        reg_email: "",
        password: ""
    }); 
    const {reg_email, password} = inputs;

    const onChange = (e) => {
        setInputs({...inputs, [e.target.name]:e.target.value});
    };

    const onSubmitForm = async (e) => {
        e.preventDefault()
        try {
            const body = {reg_email, password}
            const response = await fetch("http://localhost:4020/auth/login/parent", {
                method: "POST",
                headers: {"Content-Type": "application/json"},
                body: JSON.stringify(body)
            })
            const parseRes = await response.json();
            if(parseRes.token){
                localStorage.setItem("token", parseRes.token);
                setAuth(true)
                toast.success("Login successfully.");
            } else {
                setAuth(false);
                toast.error(parseRes);
            }

        } catch(err) {
            console.error(err.message);
        }
    }
    return (
        <Fragment>
            <h1 className="text-center my-5">Parent Login</h1>
            <form onSubmit={onSubmitForm}>
                <label>Email: </label><input type="email" name="reg_email" placeholder="email" className="form-control my-2" value={reg_email} onChange={e => onChange(e)}/>
                <label>Password: </label><input type="password" name="password" placeholder="password" className="form-control my-2" value={password} onChange={e => onChange(e)}/>
                <button data-testid="loginBtn" className="btn btn-success btn-block" color="pink">Login</button>
            </form>
        </Fragment>
    )
}

export default ParentLogin;