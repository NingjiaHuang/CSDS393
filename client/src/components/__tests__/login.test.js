import {render, screen, cleanup, fireEvent, act, getByDisplayValue, getByText, getByPlaceholderText} from '@testing-library/react';
import login_admin from '../login_admin';
import AdminLogin from '../login_admin';
import BreederLogin from '../login_breeder';
import ParentLogin from '../login_parent';
import login_breeder from '../login_breeder';
import login_parent from '../login_parent';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect';

afterEach(() =>{
    cleanup();
})


test('matches snapshot of admin login', () =>{
    const tree = renderer.create(<login_admin/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("admin email input should accept text", () => {
    const {getByLabelText} = render(<AdminLogin/>);
    const emailInputNode = getByLabelText("Email:");
    expect(emailInputNode.value).toMatch("");
    fireEvent.change(emailInputNode, {target:{value:'test@test.com'}});
    expect (emailInputNode.value).toMatch("test@test.com")
})

test("admin password input should accept text", () => {
    const {getByLabelText} = render(<AdminLogin/>);
    const passwordInputNode = getByLabelText("Password:");
    expect(passwordInputNode.value).toMatch("");
    fireEvent.change(passwordInputNode, {target:{value:'12345678'}});
    expect (passwordInputNode.value).toMatch("12345678")
})



test('matches snapshot of breeder login', () =>{
    const tree = renderer.create(<login_breeder />).toJSON();
    expect(tree).toMatchSnapshot();
})

test("breeder email input should accept text", () => {
    const {getByLabelText} = render(<BreederLogin/>);
    const emailInputNode = getByLabelText("Email:");
    expect(emailInputNode.value).toMatch("");
    fireEvent.change(emailInputNode, {target:{value:'test@test.com'}});
    expect (emailInputNode.value).toMatch("test@test.com")
})

test("breeder password input should accept text", () => {
    const {getByLabelText} = render(<BreederLogin/>);
    const passwordInputNode = getByLabelText("Password:");
    expect(passwordInputNode.value).toMatch("");
    fireEvent.change(passwordInputNode, {target:{value:'12345678'}});
    expect (passwordInputNode.value).toMatch("12345678")
})



test('matches snapshot of potential parents login', () =>{
    const tree = renderer.create(<login_parent/>).toJSON();
    expect(tree).toMatchSnapshot();
    })


test("parent email input should accept text", () => {
    const {getByLabelText} = render(<ParentLogin/>);
    const emailInputNode = getByLabelText("Email:");
    expect(emailInputNode.value).toMatch("");
    fireEvent.change(emailInputNode, {target:{value:'test@test.com'}});
    expect (emailInputNode.value).toMatch("test@test.com")
})

test("parent password input should accept text", () => {
    const {getByLabelText} = render(<ParentLogin/>);
    const passwordInputNode = getByLabelText("Password:");
    expect(passwordInputNode.value).toMatch("");
    fireEvent.change(passwordInputNode, {target:{value:'12345678'}});
    expect (passwordInputNode.value).toMatch("12345678")
})
    

describe("Admin clicks", () => {
    it('log in button should be clicked ', () => {
        const mockOnClick = jest.fn()
        const { getByTestId } = render(<AdminLogin onClick={mockOnClick()} />)
      
        const clickIndicator = getByTestId('loginBtn')

        fireEvent.click(clickIndicator)

        expect(mockOnClick).toHaveBeenCalledTimes(1)

    })
    });


describe("breeder clicks", () => {
it('log in button should be clicked ', () => {
    const mockOnClick = jest.fn()
    const { getByTestId } = render(<BreederLogin onClick={mockOnClick()} />)
    
    const clickIndicator = getByTestId('loginBtn')

    fireEvent.click(clickIndicator)

    expect(mockOnClick).toHaveBeenCalledTimes(1)

})
});


describe("Parent clicks", () => {
it('log in button should be clicked ', () => {
    const mockOnClick = jest.fn()
    const { getByTestId } = render(<ParentLogin onClick={mockOnClick()} />)
    
    const clickIndicator = getByTestId('loginBtn')

    fireEvent.click(clickIndicator)

    expect(mockOnClick).toHaveBeenCalledTimes(1)

})
});