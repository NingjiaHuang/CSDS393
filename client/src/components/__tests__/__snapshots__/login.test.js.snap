// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of admin login 1`] = `<login_admin />`;

exports[`matches snapshot of breeder login 1`] = `<login_breeder />`;

exports[`matches snapshot of potential parents login 1`] = `<login_parent />`;
